server/integration/infinispan/src/main/java/org/jboss/as/clustering/infinispan/cs/factory/DeployedCacheStoreFactory.java

```diff
@@ -35,32 +43,42 @@ public StoreConfiguration processConfiguration(StoreConfiguration storeConfigura
       if(storeConfiguration instanceof DeployedStoreConfiguration) {
          DeployedStoreConfiguration deployedConfiguration = (DeployedStoreConfiguration) storeConfiguration;
          PersistenceConfigurationBuilder replacedBuilder = deployedConfiguration.getPersistenceConfigurationBuilder();
-         DeployedCacheStoreMetadata deployedCacheStoreMetadata = deployedCacheStores.get(deployedConfiguration.getCustomStoreClassName());
 
-         if(deployedCacheStoreMetadata == null) {
-            throw new IllegalStateException("Could not find Deployed Cache metadata for " + deployedConfiguration.getCustomStoreClassName());
-         }
+         try {
+            DeployedCacheStoreMetadata deployedCacheStoreMetadata = getPromise(deployedConfiguration.getCustomStoreClassName()).get(TIMEOUT_SECONDS, TimeUnit.SECONDS);
 
-         StoreConfigurationBuilder replacedStoreBuilder = replacedBuilder.addStore(deployedCacheStoreMetadata.getStoreBuilderClass());
-         replacedStoreBuilder.fetchPersistentState(deployedConfiguration.fetchPersistentState());
-         replacedStoreBuilder.ignoreModifications(deployedConfiguration.ignoreModifications());
-         replacedStoreBuilder.preload(deployedConfiguration.preload());
-         replacedStoreBuilder.purgeOnStartup(deployedConfiguration.purgeOnStartup());
-         replacedStoreBuilder.shared(deployedConfiguration.shared());
-         replacedStoreBuilder.withProperties(deployedConfiguration.properties());
-         StoreConfiguration replacedConfiguration = (StoreConfiguration) replacedStoreBuilder.create();
+            StoreConfigurationBuilder replacedStoreBuilder = replacedBuilder.addStore(deployedCacheStoreMetadata.getStoreBuilderClass());
+            replacedStoreBuilder.fetchPersistentState(deployedConfiguration.fetchPersistentState());
+            replacedStoreBuilder.ignoreModifications(deployedConfiguration.ignoreModifications());
+            replacedStoreBuilder.preload(deployedConfiguration.preload());
+            replacedStoreBuilder.purgeOnStartup(deployedConfiguration.purgeOnStartup());
+            replacedStoreBuilder.shared(deployedConfiguration.shared());
+            replacedStoreBuilder.withProperties(deployedConfiguration.properties());
+            StoreConfiguration replacedConfiguration = (StoreConfiguration) replacedStoreBuilder.create();
 
-         return replacedConfiguration;
+            return replacedConfiguration;
+         } catch (Exception e) {
+            InfinispanLogger.ROOT_LOGGER.error("Could not get Custom Cache Store metadata (" + deployedConfiguration.getCustomStoreClassName() + ") within given time.", e);
```

Comment:
That wasn't the point, the point is catching Exception or Throwable.  I usually frown away from doing that unless I know I control the flow of the code base, which at this point we don't really.  If say we had some bug that throws a NPE it would be confusing to see a message that says it couldn't get something in time to then see a NPE as the cause.  Maybe our users are quick studies but where I worked before this would confuse people :P